# Activity QC



```{r, echo=FALSE}
library(png)
library(grid)

```


## Retained cCREs and barcodes
**Goal:** Assess cCRE and BC coverage <br>
**Input file:** activity_per_rep<br>
**Evaluated metrics:** DNA Complexity, RNA Complexity<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/retained_cCREs_and_barcodes.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/retained_cCREs_and_barcodes.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```




**Legend:** Percentage of cCREs and BCs present in the DNA and RNA quantification data compared with those observed in the association data <br>
**Interpretation:** The successful example showcases high retention of BCs and cCREs in both RNA and DNA data. The unsuccessful example showcases low retention, particularly of DNA BCs<br>


## DNA counts vs GC content

**Goal:** Assess GC content bias in PCR amplification  <br>
**Input file:** oligo_fasta, comb_df  <br>
**Evaluated metrics:** DNA Complexity <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/GC_content_bias_in_DNA.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/GC_content_bias_in_DNA.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** Each figure includes two panels with a shared x-axis showing binned GC content. The box plots showcase the number of DNA reads per cCRE. The histogram showcases the number of cCREs in each GC content bin. Data was binned in fixed sizes of 5%. <br>
**Interpretation:** In the successful example, read counts are relatively consistent across GC content, indicating PCR conditions well matched to the GC composition of most cCREs, as reflected by closely aligned peaks. The unsuccessful example showcases pronounced amplification bias and PCR conditions that are poorly optimized for the GC content of the majority of cCREs.<br>

## Activity statistic vs count ratio

**Goal:** Assess the correlation between log2(RNA/DNA) and the summary statistic of activity <br>
**Input file:** comb_df <br>
**Evaluated metrics:** DNA Complexity, RNA Complexity <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/Activity_statistic_vs_count_ratio.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L4a2/Activity_statistic_vs_count_ratio.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** A scatter plot of raw actviity values x-axis vs the activity summary statistic. Colors show density.<br>
**Interpretation:** The successful example showcases an experiment with high complexity, reflected in the high correlation between the measurements. The unsuccessful example showcases an experiment with a relatively low correlation between measurements, suggesting low complexity<br>


## Activity distribution
**Goal:** Assess activity dynamic range, noise, and statistical power <br>
**Input file:** comb_df<br>
**Evaluated metrics:** DNA Complexity, RNA Complexity:<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:32%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/activity_distribution.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:32%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L4a2/activity_distribution.svg",
        style = "width:100%; display:block;"
      )
    ),
    htmltools::tags$div(
      style = "width:32%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/activity_distribution.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** A histogram of log2(RNA/DNA) of the tested cCREs, with significantly active cCREs marked in red. <br>
**Interpretation:** The successful example showcases a broad dynamic range, with most of the highly active cCREs detected as significantly active. The 1st unsuccessful example showcases a problem in the detection of activity, either because of high noise levels or the experiment being underpowered. The 2nd unsuccessful example showcases silencing, with cCREs that are unlikely to be truly active detected as significantly active<br>


## Differential activity distribution

**Goal:** Assess differential activity dynamic range, noise, and statistical power <br>
**Input file:** comparative_df <br>
**Evaluated metrics:** DNA Complexity, RNA Complexity<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/Differential_activity_distribution.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/Differential_activity_distribution.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** A histogram of the log2(RNA/DNA) difference between the two alleles of a cCRE, with significantly differentially active cCREs marked in red. <br>
**Interpretation:** The successful example showcases a broad dynamic range of differential activity, with high fold-changes detected as significantly differential active. The unsuccessful example showcases underdetection of differential activity or high noise levels.<br>


## P-value distribution
**Goal:** Assess statistical power <br>
**Input file:** comb_df<br>
**Evaluated metrics:** DNA Complexity, RNA Complexity, Reproducibility, Dynamic range<br>
```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/P_value_distribution.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/P_value_distribution.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** A qq plot of the p-values assigned to each cCRE activity level compared to the expected p-values generated by a uniform distribution <br>
**Interpretation:** The successful example showcases sufficient statistical power, reflected in a downward deviation from the expected distribution, with many low p-values. The unsuccessful example showcases an upward deviation, reflecting an inflation of p-values.<br>

## Volcano plot - FC vs Pval

**Goal:** Assess statistical power of differential activity detection <br>
**Input file:** comparative_df<br>
**Evaluated metrics:** DNA Complexity, RNA Complexity <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/Differential_activity_volcano_zoom.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/Differential_activity_volcano_zoom.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** A volcano plot of logFC vs FDR. Dashed line shows FDR = 0.05.  <br>
**Interpretation:** The successful example showcases sufficient statistical power to detect differential activity. cCRES with large absolute(logFC) are detected as differentially active. The unsuccessful example showcases many cCREs with relatively large absolute(logFC) that are not detected as differentially active, suggesting low power or high noise levels.<br>

## BC retention by DNA/RNA sequencing depth

**Goal:** Assess whether sequencing depth is sufficient <br>
**Input file:** downsampling_ratio_path <br>
**Evaluated metrics:** DNA Complexity, RNA Complexity <br>
```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/BC_retention_by_DNA_RNA_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/BC_retention_by_DNA_RNA_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** The relationship between sequencing depth and the percentage of retained BCs. Data is downsampled in order to assess whether we have reached saturation in the percentage of retained BCs. Then, data is extrapolated to predict whether additional sequencing would help improve the results. <br>
**Interpretation:** -	3.8: In the following point, please update the numbers: Interpretation: The successful example showcases sufficient sequencing depth: 3-fold more sequencing will only increase BC DNA retention from ~87% to 94% (1.08-fold), and BC RNA from 92% to 95% (1.03-fold). The unsuccessful example showcases suboptimal RNA sequencing depth: additional sequencing is predicted to increase the number of retained BCs by 1.33-fold (from 24% to 32%).<br>


## cCRE retention by DNA/RNA sequencing depth

**Goal:** Assess whether sequencing depth is sufficient <br>
**Input file:** downsampling_ratio_path <br>
**Evaluated metrics:** DNA Complexity, RNA Complexity<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/cCRE_retention_by_DNA_RNA_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/cCRE_retention_by_DNA_RNA_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** The relationship between sequencing depth and the percentage of retained cCREs. Data is downsampled in order to assess whether we have reached saturation in the percentage of retained cCREs. Then, data is extrapolated to predict whether additional sequencing would help improve the results. <br>
**Interpretation:** Interpretation: The successful example showcases sufficient sequencing depth: 3-fold more sequencing will not increase cCRE retention. The unsuccessful example showcases suboptimal RNA sequencing depth: additional sequencing is predicted to increase the percentage of retained cCREs from 66% to 72%.<br>



## Activity by sequencing depth
**Goal:** Assess if sequencing depth is sufficient <br>
**Input file:** downsampling_activity_path<br>
**Evaluated metrics:** DNA Complexity, RNA Complexity<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/Activity_by_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/Activity_by_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** The relationship between sequencing depth and the percentage of active cCREs. Data is downsampled in order to assess whether we have reached saturation in the percentage of active cCREs. Then, data is extrapolated to predict whether additional sequencing would help improve the results.  <br>
**Interpretation:** The successful example showcases sufficient sequencing depth: the percentage of active cCREs has reached a plateau, and more sequencing is not expected to affect it. The unsuccessful example showcases suboptimal RNA sequencing depth: additional sequencing will likely affect the percentage of active cCREs.<br>


## Cumulative RNA reads
**Goal:** Assess jackpotting <br>
**Input file:** comb_df<br>
**Evaluated metrics:** RNA Complexity, Dynamic range<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/cumulative_RNA_reads.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/d2Osteoblast_spiking_oligos/cumulative_RNA_reads.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** A cumulative distribution showcasing the fraction of RNA reads from each cCRE. cCREs are sorted by their RNA read counts. <br>
**Interpretation:** The successful example showcases no jackpotting – the RNA reads come from many cCREs. The unsuccessful example showcases an experiment where most of the RNA reads come from just a few cCREs.<br>




## Sample clustering
**Goal:** Assess reproducibility between samples <br>
**Input file:** cDNA_reads_by_cell_type<br>
**Evaluated metrics:** Reproducibility<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/PCA_similarity_between_samples.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/thylacine_biorxiv_Gallego_Romero/PCA_similarity_between_samples.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** Principal component analysis (PCA) on the activity of cCREs in replicates from two cell lines. <br>
**Interpretation:** In the successful example, PC1 separates between cell types. In the unsuccessful example, variation between replicates is as large as that between cell types, and one of the replicates of cell type 2 clusters closely with cell type 1.<br>



## Correlation between replicates
**Goal:** Assess reproducibility between replicates<br>
**Input file:** activity_per_rep<br>
**Evaluated metrics:** Reproducibility, Dynamic range<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/thylacine_biorxiv_Gallego_Romero/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L4a2/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** Scatter plot of the log2(RNA/DNA) values of each cCRE across two experimental replicates. Each data point is a hexagon representing the local density of cCREs. <br>
**Interpretation:** The successful example showcases a high correlation between replicates in more active cCREs. The unsuccessful example showcases no correlation<br>



## Correlation of differential activity between replicates

**Goal:** Assess reproducibility between replicates <br>
**Input file:** allelic_pairs_replicates_df<br>
**Evaluated metrics:** Reproducibility, Dynamic range <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/diff_activity_corr_reps_hexbin_w_bar.svg",
                          "external_figures/humanMPRA_L4a2/diff_activity_corr_reps_hexbin_w_bar.svg"))
```

**Legend:** Scatter plot of the log2(RNA/DNA) allelic difference values for each cCRE in two replicates. Each data point is a hexagon representing the local density of cCREs. <br>
**Interpretation:** The successful example showcases concordance in differential activity between repliactes. The unsuccessful example showcases little concordance.<br>






## Replicability by activity
**Goal:** Assess the correlation in activity between replicates in active vs non-active cCREs <br>
**Input file:** activity_per_rep<br>
**Evaluated metrics:** Reproducibility <br>
```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/Variation_at_various_activity_levels.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/Variation_at_various_activity_levels.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** Correlation between replicates as a function of average activity [log₂(RNA/DNA)], shown across fixed bins of width 1. <br>
**Interpretation:** In the successful example, more active cCREs showcase a higher correlation, which further improves with higher activity level. The unsuccessful example showcases a low correlation, which remains poor in active cCREs.<br>






## Correlation between replicates (controls)
**Goal:** Assess reproducibility between replicates in positive and negative controls <br>
**Input file:** comb_df<br>
**Evaluated metrics:** Reproducibility, Dynamic range<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** Scatter plot of the log2(RNA/DNA) values of each cCRE across two experimental replicates. Positive controls are marked in green, and negative controls are marked in red. <br>
**Interpretation:** The successful example showcases a high correlation between positive controls and a low correlation between negative controls. The unsuccessful example showcases low correlations in both.<br>




## Cross-validation: allelic pairs

**Goal:** Assess reproducibility between allelic pairs <br>
**Input file:** allelic_pairs_df <br>
**Evaluated metrics:** Reproducibility <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/allelic_pairs_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/allelic_pairs_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** Log2(RNA/DNA) values of the two alleles of a cCRE. Each data point is a hexagon representing the local density of cCREs.<br>
**Interpretation:** When allelic pairs differ by only a few nucleotides, their regulatory activity should be highly correlated. The successful example showcases a strong correlation between allelic pairs. The unsuccessful example showcases little to no correlation.<br>





## Cross-validaiton: cell types

**Goal:** Assess reproducibility between different cell types<br>
**Input file:** cell_types_df <br>
**Evaluated metrics:** Reproducibility<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_NPC/cell_types_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/cell_types_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** Log2(RNA/DNA) values of shared control sequences across two cell types. Each data point is a hexagon representing the local density of cCREs. <br>
**Interpretation:** Although cell types differ in their trans environments, many regulatory networks are shared, particularly between closely related cell types. As a result, regulatory sequences often exhibit similar activity across cell types. This pattern is evident in the successful example, which shows a strong correlation between cell types. The unsuccessful example, on the other hand, showcases no correlation.<br>


## Minimizing noise 
**Goal:** Remove outlier barcodes and optimize minimum DNA count threshold to improve reproducibility<br>
**Input file:** different_std_threshold_analysis<br>
**Evaluated metrics:** Reproducibility<br>

```{r echo=FALSE, fig.show="hold", out.width="100%"}
knitr::include_graphics("external_figures/modern_humanMPRA_Hob/minimizing_noise_hexbin.svg",
                          )
```
**Legend:** Each panel shows the correlation between replicates across different threshold of outlier removal (horizontal), and minimum DNA counts per cCRE (vertical). Each data point is a hexagon representing the local density of cCREs.<br>
**Interpretation:** The examples show an increase in correlations between replicates with increasing DNA cutoffs and outlier BC removal<br>


## Reproducibility by sequencing depth

**Goal:** Assess whether additional sequencing will improve reproducibility between replicates <br>
**Input file:** downsampling_activity_path, downsampling_ratio_path <br>
**Evaluated metrics:** Reproducibility <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/Reproducibility_by_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/Reproducibility_by_sequencing_depth.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** The correlation between replicates as a function of the sequencing depth. The x-axis shows increasing levels of downsampling of the current data. <br>
**Interpretation:** In the successful example, additional sequencing is expected to slightly improve reproducibility, whereas in the unsuccessful example it is not.<br>


## RNA vs DNA
**Goal:** Assess true activity in the experiment <br>
**Input file:** comb_df<br>
**Evaluated metrics:** Dynamic Range<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:32%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:32%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L4a2/RNA_DNA_ratio_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    ),
    htmltools::tags$div(
      style = "width:32%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/RNA_DNA_ratio_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** RNA vs. DNA counts for each cCRE, normalized to sequencing depth. Each data point is a hexagon representing the local density of cCREs. <br>
**Interpretation:** The successful example showcases many sequences with high RNA-to-DNA ratios, reflected by points above the diagonal. The 1st unsuccessful example showcases little to no activity, with points clustering along the diagonal, and the 2nd unsuccessful example showcases silencing, reflected by many points falling below the diagonal.<br>




## Activity of controls
**Goal:** Assess the dynamic range of activity<br>
**Input file:** comb_df<br>
**Evaluated metrics:** Dynamic Range<br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/Control_activity_boxplots.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/Max_MPRA_run2/Control_activity_boxplots.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** Box plots of activity levels for positive and negative controls, as well as cCREs. <br>
**Interpretation:** In the successful example, positive controls show high activity while negative controls show low activity. The unsuccessful example shows similar activity levels for positive and negative controls.<br>


## Genomic annotations

**Goal:** Assess concordance with endogenous signals of active chromatin marks<br>
**Input file:** screen_df <br>
**Evaluated metrics:** Concordance with endogenous signals <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/Genomic_annotations.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/Genomic_annotations.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** A stacked bar plot of cCRE overlap with active chromatin marks (ENCODE SCREEN elements). cCREs were grouped into six groups based on their activity levels. <br>
**Interpretation:** The successful example showcases increased overlap with active chromatin marks as activity increases. The unsuccessful example shows no such increase<br>


## Proximity to TSS


**Goal:** Assess concordance with endogenous locations of cCREs <br>
**Input file:** tss_df<br>
**Evaluated metrics:** Concordance with endogenous signals <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/PMID_38766054_Reilly/Proximity_to_TSS.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/Proximity_to_TSS.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```


**Legend:** Box plots of cCRE distance to the closest TSS. cCREs were grouped into six groups based on their activity levels. <br>
**Interpretation:** The successful example showcases that more active cCREs tend to be closer to a TSS. The unsuccessful example shows no such trend<br>




## AI predictions vs activity

**Goal:** Assess the correlation between AI predictions of activity and MPRA results <br>
**Input file:** AI_df <br>
**Evaluated metrics:** Concordance with endogenous signals <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/AI_predictions_vs_activity_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_activity_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```



**Legend:** Experimentally measured activity vs AI-predicted activity. Each data point is a hexagon representing the local density of cCREs.<br>
**Interpretation:** The successful example showcases a high correlation, and the unsuccessful example showcases little correlation<br>



## AI predictions vs differential activity

**Goal:** Assess the correlation between AI predictions of differential activity and MPRA results <br>
**Input file:** AI_comparative_df <br>
**Evaluated metrics:** Concordance with endogenous signals <br>

```{r echo=FALSE}
htmltools::tagList(
  htmltools::tags$div(
    style = "display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;",

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_up.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src = "external_figures/modern_humanMPRA_Hob/AI_predictions_vs_differential_activity_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    ),

    htmltools::tags$div(
      style = "width:49%;",
      htmltools::tags$div(
        style = "text-align:center; margin-bottom:6px;",
        htmltools::tags$img(
          src = "external_figures/thumbs_down.svg",
          style = "width:60px; display:inline-block;"
        )
      ),
      htmltools::tags$img(
        src =   "external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_differential_activity_hexbin_w_bar.svg",
        style = "width:100%; display:block;"
      )
    )
  )
)

```




**Legend:** Experimentally measured differential activity vs AI-predicted differential activity. Each data point is a hexagon representing the local density of cCREs. <br>
**Interpretation:** The successful example showcases some correlation, and the unsuccessful example showcases no correlation.<br>






