# Activity QC



```{r, echo=FALSE}
library(png)
library(grid)

```

```{r, echo=FALSE}
working_dir="/home/labs/davidgo/Collaboration/MPRA_QC_pipeline/activity/output"
output_dir=sprintf("%s/test_QC",working_dir)
verbose = TRUE

print_examples <- function(good_example, bad_example, bad_example2 = NULL) {
  if (exists("verbose", envir = .GlobalEnv) && get("verbose", envir = .GlobalEnv)) {
    cat("Good example:", good_example, "\n")
    cat("Bad example:", bad_example, "\n")
    if (!is.null(bad_example2)) {
      cat("Bad example 2:", bad_example2, "\n")
    }
  }
}

plot_single <- function(example) {
  # Start new page
  grid::grid.newpage()
  
  # Draw the image directly (no layout needed for one plot)
  grid::grid.raster(example, interpolate = FALSE)
}


plot_side_by_side <- function(good_example, bad_example) {
  # Start new page and define layout
  grid::grid.newpage()
  grid::pushViewport(grid::viewport(layout = grid::grid.layout(1, 2)))
  
  # Helper to specify location in layout
  vplayout <- function(row, col) grid::viewport(layout.pos.row = row, layout.pos.col = col)
  
  # Draw both images in the layout without using print()
  grid::pushViewport(vplayout(1, 1))
  grid::grid.raster(good_example, interpolate = FALSE)
  grid::popViewport()
  
  grid::pushViewport(vplayout(1, 2))
  grid::grid.raster(bad_example, interpolate = FALSE)
  grid::popViewport()
}


plot_three_side_by_side <- function(good_example, bad_example1, bad_example2) {
  # Start new page and define layout
  grid::grid.newpage()
  grid::pushViewport(grid::viewport(layout = grid::grid.layout(1, 3)))
  
  # Helper to specify location in layout
  vplayout <- function(row, col) grid::viewport(layout.pos.row = row, layout.pos.col = col)
  
  # Draw images in the layout without using print()
  grid::pushViewport(vplayout(1, 1))
  grid::grid.raster(good_example, interpolate = FALSE)
  grid::popViewport()
  
  grid::pushViewport(vplayout(1, 2))
  grid::grid.raster(bad_example1, interpolate = FALSE)
  grid::popViewport()
  
  grid::pushViewport(vplayout(1, 3))
  grid::grid.raster(bad_example2, interpolate = FALSE)
  grid::popViewport()
}

library(png)
library(grid)

plot_all_MPRA <- function(working_dir, analysis_name, ncol = 3) {
  # working_dir: path where all MPRA folders are located
  # analysis_name: file name of the PNG image to plot (e.g. "PCA_similarity_between_samples.png")
  # ncol: how many figures per row
  
  # list all MPRA library folders
  mpra_libraries <- list.dirs(working_dir, recursive = FALSE, full.names = FALSE)
  
  # collect available images
  images <- list()
  for (lib in mpra_libraries) {
    img_path <- file.path(working_dir, lib, analysis_name)
    if (file.exists(img_path)) {
      images[[lib]] <- readPNG(img_path, native = TRUE)
    }
  }
  
  if (length(images) == 0) {
    message("No images found for ", analysis_name)
    return(NULL)
  }
  
  n <- length(images)
  nrow <- ceiling(n / ncol)
  
  grid.newpage()
  pushViewport(viewport(layout = grid.layout(nrow, ncol)))
  
  i <- 1
  for (lib in names(images)) {
    row <- ceiling(i / ncol)
    col <- i - (row - 1) * ncol
    pushViewport(viewport(layout.pos.row = row, layout.pos.col = col))
    grid.raster(images[[lib]], interpolate = FALSE)
    grid.text(lib, y = unit(0.05, "npc"), gp = gpar(col = "black", fontsize = 10))
    popViewport()
    i <- i + 1
  }
}


```




## Retained cCREs and barcodes
**Goal: Assesses cCRE and BC coverage** <br>
**Input file: activity_per_rep**<br>
**Evaluated metrics: DNA Complexity, RNA Complexity**:<br>


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/retained_cCREs_and_barcodes.svg",
                          "external_figures/Max_MPRA_run2/retained_cCREs_and_barcodes.svg"))
```


**Legend: Percentage of cCREs and BCs present in the DNA and RNA quantification data compared with those observed in the association data** <br>
**Interpretation:**<br>


## DNA counts vs GC content

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/GC_content_bias_in_DNA.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/GC_content_bias_in_DNA.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Activity statistic vs count ratio

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/Activity_statistic_vs_count_ratio.svg",
                          "external_figures/humanMPRA_L4a2/Activity_statistic_vs_count_ratio.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Differential activity statistic vs count ratio

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>


## Activity distribution
**Goal: Assesses activity dynamic range, noise, and statistical power** <br>
**Input file: quantification table (comb_df)**<br>
**Evaluated metrics: DNA Complexity, RNA Complexity**:<br>

```{r echo=FALSE, fig.show="hold", out.width=rep("32%", 3)}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/activity_distribution.svg",
                          "external_figures/humanMPRA_L4a2/activity_distribution.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/activity_distribution.svg"
                          ))
```
**Legend:** <br>
**Interpretation:**<br>


## Differential activity distribution

**Goal: Assesses differential activity dynamic range, noise, and statistical power** <br>
**Input file: comparative_df** <br>
**Evaluated metrics: DNA Complexity, RNA Complexity** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/Differential_activity_distribution.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/Differential_activity_distribution.svg"))
```

**Legend:** <br>
**Interpretation:**<br>


## P-value distribution
**Goal: Inferring the power of the statistical test** <br>
**Input file: quantification table (comb_df)**<br>
**Evaluated metrics: DNA Complexity, RNA Complexity, Reproducibility, Dynamic range**:<br>


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/P_value_distribution.svg",
                          "external_figures/Max_MPRA_run2/P_value_distribution.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Volcano plot - FC vs Pval

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/Differential_activity_volcano_zoom.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/Differential_activity_volcano_zoom.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## BC retention by DNA/RNA sequencing depth

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/BC_retention_by_DNA_RNA_sequencing_depth.svg",
                          "external_figures/Max_MPRA_run2/BC_retention_by_DNA_RNA_sequencing_depth.svg"))
```

**Legend:** <br>
**Interpretation:**<br>


## cCRE retention by DNA/RNA sequencing depth

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/cCRE_retention_by_DNA_RNA_sequencing_depth.svg",
                          "external_figures/Max_MPRA_run2/cCRE_retention_by_DNA_RNA_sequencing_depth.svg"))
```

**Legend:** <br>
**Interpretation:**<br>



## Activity by sequencing depth
**Goal: uses downsampling of sequencing reads to assess if sequencing depth is sufficient** <br>
**Input file: Downsampling activity data frames**<br>
**Evaluated metrics: DNA Complexity, RNA Complexity**<br>

```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/Activity_by_sequencing_depth.svg",
                          "external_figures/modern_humanMPRA_NPC/Activity_by_sequencing_depth.svg"))
```
**Legend:** <br>
**Interpretation:**<br>


## Cumulative RNA reads
**Goal: assesses jackpotting** <br>
**Input file: quantification table (comb_df)**<br>
**Evaluated metrics: DNA Complexity, RNA Complexity**:<br>


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/cumulative_RNA_reads.svg",
                          "external_figures/d2Osteoblast_spiking_oligos/cumulative_RNA_reads.svg"))
```

**Legend:** <br>
**Interpretation:**<br>




## Sample clustering
**Goal: Assesses reproducibility between samples** <br>
**Input file: cDNA_reads_by_cell_type**<br>
**Evaluated metrics: Reproducibility**:<br>

mention in the bookdown: the importance of the percentage explained by the 1st and 2nd PCs. 


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/PCA_similarity_between_samples.svg",
                          "external_figures/thylacine_biorxiv_Gallego_Romero/PCA_similarity_between_samples.svg"))
```

**Legend: Principal component analysis (PCA) on samples** <br>
**Interpretation:**<br>



## Correlation between replicates
**Goal: assesses reproducibility between replicates** <br>
**Input file: activity_per_rep**<br>
**Evaluated metrics: Reproducibility**:<br>

```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/thylacine_biorxiv_Gallego_Romero/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg",
                          "external_figures/humanMPRA_L4a2/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg"
                          ))
```
**Legend:** <br>
**Interpretation:**<br>



## Correlation of differential activity between replicates

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>






## Replicability by activity
**Goal: assesses the correlation in activity between replicates in active vs non-active cCREs** <br>
**Input file:**<br>
**Evaluated metrics: Reproducibility **<br>


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/Variation_at_various_activity_levels.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/Variation_at_various_activity_levels.svg"))
```
**Legend:** <br>
**Interpretation:**<br>






## Correlation between replicates (controls)
**Goal: analysis assesses reproducibility between replicates in positive and negative controls** <br>
**Input file: quantification table (comb_df)**<br>
**Evaluated metrics: Reproducibility**<br>

```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg",
                          "external_figures/Max_MPRA_run2/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg"))
```
**Legend:** <br>
**Interpretation:**<br>




## Cross-validation:allelic pairs

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/allelic_pairs_hexbin_w_bar.svg",
                          "external_figures/Max_MPRA_run2/allelic_pairs_hexbin_w_bar.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Correlation of differential activity between overlapping sequences

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Cross-validaiton: cell types

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Correlation of differential activity between cell types

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Cross-validation: experiments

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Correlation of differential activity between experiments

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>

## Replicability across experiments

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/placeholder.svg",
                          "external_figures/placeholder.svg"))
```

**Legend:** <br>
**Interpretation:**<br>




## Minimizing noise 
**Goal: Removing outlier BCs and optimizing the threshold of minimum DNA counts to increase reproducibility** <br>
**Input file: **<br>
**Evaluated metrics: Reproducibility**:<br>

```{r echo=FALSE, fig.show="hold", out.width="100%"}
knitr::include_graphics("external_figures/modern_humanMPRA_Hob/minimizing_noise_hexbin.svg",
                          )
```
**Legend:** <br>
**Interpretation:**<br>


## Reproducibility by sequencing depth

**Goal: Assesses whether additional sequencing will improve reproducibility between replicates** <br>
**Input file: Activity down sampling files** <br>
**Evaluated metrics: Reproducibility** <br>


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/Reproducibility_by_sequencing_depth.svg",
                          "external_figures/Max_MPRA_run2/Reproducibility_by_sequencing_depth.svg"))
```

**Legend:** <br>
**Interpretation:**<br>


## RNA vs DNA
**Goal: Assesses if there is true activity in the experiment** <br>
**Input file:**<br>
**Evaluated metrics: Dynamic Range**<br>


```{r echo=FALSE, fig.show="hold", out.width=rep("32%", 3)}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_hexbin_w_bar.svg",
                          "external_figures/humanMPRA_L4a2/RNA_DNA_ratio_hexbin_w_bar.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/RNA_DNA_ratio_hexbin_w_bar.svg"
                          ))
```

**Legend:** <br>
**Interpretation:**<br>




## Activity of controls
**Goal: assesses the dynamic range of activity** <br>
**Input file: quantification table (comb_df)**<br>
**Evaluated metrics: Dynamic Range**:<br>


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/Control_activity_boxplots.svg",
                          "external_figures/Max_MPRA_run2/Control_activity_boxplots.svg"
                          ))
```
**Legend:** <br>
**Interpretation:**<br>


## Genomic annotations

**Goal: This analysis assesses concordance with endogenous signals of active chromatin marks** <br>
**Input file: Genomic annotation file** <br>
**Evaluated metrics: Dynamic Range** <br>


```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/Genomic_annotations.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/Genomic_annotations.svg"
                          ))
```

**Legend:** <br>
**Interpretation:**<br>




## Proximity to TSS


**Goal: is analysis assesses concordance with endogenous locations of cCREs** <br>
**Input file: Distance to TSS file** <br>
**Evaluated metrics: Dynamic Range** <br>

```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/PMID_38766054_Reilly/Proximity_to_TSS.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/Proximity_to_TSS.svg"
                          ))
```

**Legend:** <br>
**Interpretation:**<br>




## AI predictions vs activity

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/AI_predictions_vs_activity_hexbin_w_bar.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_activity_hexbin_w_bar.svg"))
```

**Legend:** <br>
**Interpretation:**<br>



## AI predictions vs differential activity

**Goal: ** <br>
**Input file: ** <br>
**Evaluated metrics: ** <br>



```{r echo=FALSE, fig.show="hold", out.width=c("49%","49%")}
knitr::include_graphics(c("external_figures/modern_humanMPRA_Hob/AI_predictions_vs_differential_activity_hexbin_w_bar.svg",
                          "external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_differential_activity_hexbin_w_bar.svg"))
```

**Legend:** <br>
**Interpretation:**<br>






