<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Activity QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)</title>
  <meta name="description" content="A bookdown guide containing benchmarked quality-control examples for successful and unsuccessful MPRA experiments." />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Activity QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A bookdown guide containing benchmarked quality-control examples for successful and unsuccessful MPRA experiments." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Activity QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)" />
  
  <meta name="twitter:description" content="A bookdown guide containing benchmarked quality-control examples for successful and unsuccessful MPRA experiments." />
  

<meta name="author" content="Simon Fishilevich, Omer Ronen, Nadav Mishol, Ryder Easterlin, Nitzan Haim, Katharina Lange, Nadav Ahituv, Irene Gallego Romero, Fumitaka Inoue, Martin Kircher, Michael I. Love, Max Schubach, and David Gokhman" />


<meta name="date" content="2026-02-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="associations-qc.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#abbreviations"><i class="fa fa-check"></i>Abbreviations</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i>Usage</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#scripts"><i class="fa fa-check"></i>Scripts</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html"><i class="fa fa-check"></i><b>1</b> A guide for running the analyses notebook</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#associations"><i class="fa fa-check"></i><b>1.1</b> Associations</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#script"><i class="fa fa-check"></i><b>1.1.1</b> Script</a></li>
<li class="chapter" data-level="1.1.2" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#input"><i class="fa fa-check"></i><b>1.1.2</b> Input</a></li>
<li class="chapter" data-level="1.1.3" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#output"><i class="fa fa-check"></i><b>1.1.3</b> Output</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#rna-and-dna-quantification"><i class="fa fa-check"></i><b>1.2</b> RNA and DNA quantification</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#script-1"><i class="fa fa-check"></i><b>1.2.1</b> Script</a></li>
<li class="chapter" data-level="1.2.2" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#input-1"><i class="fa fa-check"></i><b>1.2.2</b> Input</a></li>
<li class="chapter" data-level="1.2.3" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#output-1"><i class="fa fa-check"></i><b>1.2.3</b> Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="associations-qc.html"><a href="associations-qc.html"><i class="fa fa-check"></i><b>2</b> Associations QC</a>
<ul>
<li class="chapter" data-level="2.1" data-path="associations-qc.html"><a href="associations-qc.html#ccres-per-bc"><i class="fa fa-check"></i><b>2.1</b> cCREs per BC</a></li>
<li class="chapter" data-level="2.2" data-path="associations-qc.html"><a href="associations-qc.html#reads-per-association"><i class="fa fa-check"></i><b>2.2</b> Reads per association</a></li>
<li class="chapter" data-level="2.3" data-path="associations-qc.html"><a href="associations-qc.html#bcs-per-ccre"><i class="fa fa-check"></i><b>2.3</b> BCs per cCRE</a></li>
<li class="chapter" data-level="2.4" data-path="associations-qc.html"><a href="associations-qc.html#retained-ccres"><i class="fa fa-check"></i><b>2.4</b> Retained cCREs</a></li>
<li class="chapter" data-level="2.5" data-path="associations-qc.html"><a href="associations-qc.html#ccre-retention-by-sequencing-depth"><i class="fa fa-check"></i><b>2.5</b> cCRE retention by sequencing depth</a></li>
<li class="chapter" data-level="2.6" data-path="associations-qc.html"><a href="associations-qc.html#bcs-per-ccre-by-sequencing-depth"><i class="fa fa-check"></i><b>2.6</b> BCs per cCRE by sequencing depth</a></li>
<li class="chapter" data-level="2.7" data-path="associations-qc.html"><a href="associations-qc.html#pcr-bias---gc"><i class="fa fa-check"></i><b>2.7</b> PCR bias - GC</a></li>
<li class="chapter" data-level="2.8" data-path="associations-qc.html"><a href="associations-qc.html#pcr-bias---g-stretches"><i class="fa fa-check"></i><b>2.8</b> PCR bias - G-stretches</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="activity-qc.html"><a href="activity-qc.html"><i class="fa fa-check"></i><b>3</b> Activity QC</a>
<ul>
<li class="chapter" data-level="3.1" data-path="activity-qc.html"><a href="activity-qc.html#retained-ccres-and-barcodes"><i class="fa fa-check"></i><b>3.1</b> Retained cCREs and barcodes</a></li>
<li class="chapter" data-level="3.2" data-path="activity-qc.html"><a href="activity-qc.html#dna-counts-vs-gc-content"><i class="fa fa-check"></i><b>3.2</b> DNA counts vs GC content</a></li>
<li class="chapter" data-level="3.3" data-path="activity-qc.html"><a href="activity-qc.html#activity-statistic-vs-count-ratio"><i class="fa fa-check"></i><b>3.3</b> Activity statistic vs count ratio</a></li>
<li class="chapter" data-level="3.4" data-path="activity-qc.html"><a href="activity-qc.html#activity-distribution"><i class="fa fa-check"></i><b>3.4</b> Activity distribution</a></li>
<li class="chapter" data-level="3.5" data-path="activity-qc.html"><a href="activity-qc.html#differential-activity-distribution"><i class="fa fa-check"></i><b>3.5</b> Differential activity distribution</a></li>
<li class="chapter" data-level="3.6" data-path="activity-qc.html"><a href="activity-qc.html#p-value-distribution"><i class="fa fa-check"></i><b>3.6</b> P-value distribution</a></li>
<li class="chapter" data-level="3.7" data-path="activity-qc.html"><a href="activity-qc.html#volcano-plot---fc-vs-pval"><i class="fa fa-check"></i><b>3.7</b> Volcano plot - FC vs Pval</a></li>
<li class="chapter" data-level="3.8" data-path="activity-qc.html"><a href="activity-qc.html#bc-retention-by-dnarna-sequencing-depth"><i class="fa fa-check"></i><b>3.8</b> BC retention by DNA/RNA sequencing depth</a></li>
<li class="chapter" data-level="3.9" data-path="activity-qc.html"><a href="activity-qc.html#ccre-retention-by-dnarna-sequencing-depth"><i class="fa fa-check"></i><b>3.9</b> cCRE retention by DNA/RNA sequencing depth</a></li>
<li class="chapter" data-level="3.10" data-path="activity-qc.html"><a href="activity-qc.html#activity-by-sequencing-depth"><i class="fa fa-check"></i><b>3.10</b> Activity by sequencing depth</a></li>
<li class="chapter" data-level="3.11" data-path="activity-qc.html"><a href="activity-qc.html#cumulative-rna-reads"><i class="fa fa-check"></i><b>3.11</b> Cumulative RNA reads</a></li>
<li class="chapter" data-level="3.12" data-path="activity-qc.html"><a href="activity-qc.html#sample-clustering"><i class="fa fa-check"></i><b>3.12</b> Sample clustering</a></li>
<li class="chapter" data-level="3.13" data-path="activity-qc.html"><a href="activity-qc.html#correlation-between-replicates"><i class="fa fa-check"></i><b>3.13</b> Correlation between replicates</a></li>
<li class="chapter" data-level="3.14" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-replicates"><i class="fa fa-check"></i><b>3.14</b> Correlation of differential activity between replicates</a></li>
<li class="chapter" data-level="3.15" data-path="activity-qc.html"><a href="activity-qc.html#replicability-by-activity"><i class="fa fa-check"></i><b>3.15</b> Replicability by activity</a></li>
<li class="chapter" data-level="3.16" data-path="activity-qc.html"><a href="activity-qc.html#correlation-between-replicates-controls"><i class="fa fa-check"></i><b>3.16</b> Correlation between replicates (controls)</a></li>
<li class="chapter" data-level="3.17" data-path="activity-qc.html"><a href="activity-qc.html#cross-validation-allelic-pairs"><i class="fa fa-check"></i><b>3.17</b> Cross-validation: allelic pairs</a></li>
<li class="chapter" data-level="3.18" data-path="activity-qc.html"><a href="activity-qc.html#cross-validaiton-cell-types"><i class="fa fa-check"></i><b>3.18</b> Cross-validaiton: cell types</a></li>
<li class="chapter" data-level="3.19" data-path="activity-qc.html"><a href="activity-qc.html#minimizing-noise"><i class="fa fa-check"></i><b>3.19</b> Minimizing noise</a></li>
<li class="chapter" data-level="3.20" data-path="activity-qc.html"><a href="activity-qc.html#reproducibility-by-sequencing-depth"><i class="fa fa-check"></i><b>3.20</b> Reproducibility by sequencing depth</a></li>
<li class="chapter" data-level="3.21" data-path="activity-qc.html"><a href="activity-qc.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.21</b> RNA vs DNA</a></li>
<li class="chapter" data-level="3.22" data-path="activity-qc.html"><a href="activity-qc.html#activity-of-controls"><i class="fa fa-check"></i><b>3.22</b> Activity of controls</a></li>
<li class="chapter" data-level="3.23" data-path="activity-qc.html"><a href="activity-qc.html#genomic-annotations"><i class="fa fa-check"></i><b>3.23</b> Genomic annotations</a></li>
<li class="chapter" data-level="3.24" data-path="activity-qc.html"><a href="activity-qc.html#proximity-to-tss"><i class="fa fa-check"></i><b>3.24</b> Proximity to TSS</a></li>
<li class="chapter" data-level="3.25" data-path="activity-qc.html"><a href="activity-qc.html#ai-predictions-vs-activity"><i class="fa fa-check"></i><b>3.25</b> AI predictions vs activity</a></li>
<li class="chapter" data-level="3.26" data-path="activity-qc.html"><a href="activity-qc.html#ai-predictions-vs-differential-activity"><i class="fa fa-check"></i><b>3.26</b> AI predictions vs differential activity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="activity-qc" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Activity QC<a href="activity-qc.html#activity-qc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="retained-ccres-and-barcodes" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Retained cCREs and barcodes<a href="activity-qc.html#retained-ccres-and-barcodes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess cCRE and BC coverage <br>
<strong>Input file:</strong> activity_per_rep<br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/retained_cCREs_and_barcodes.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/retained_cCREs_and_barcodes.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Percentage of cCREs and BCs present in the DNA and RNA quantification data compared with those observed in the association data <br>
<strong>Interpretation:</strong> The successful example showcases high retention of BCs and cCREs in both RNA and DNA data. The unsuccessful example showcases low retention, particularly of DNA BCs<br></p>
</div>
<div id="dna-counts-vs-gc-content" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> DNA counts vs GC content<a href="activity-qc.html#dna-counts-vs-gc-content" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess GC content bias in PCR amplification <br>
<strong>Input file:</strong> oligo_fasta, comb_df <br>
<strong>Evaluated metrics:</strong> DNA Complexity <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/GC_content_bias_in_DNA.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/GC_content_bias_in_DNA.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Each figure includes two panels with a shared x-axis showing binned GC content. The box plots showcase the number of DNA reads per cCRE. The histogram showcases the number of cCREs in each GC content bin. Data was binned in fixed sizes of 5%. <br>
<strong>Interpretation:</strong> In the successful example, read counts are relatively consistent across GC content, indicating PCR conditions well matched to the GC composition of most cCREs, as reflected by closely aligned peaks. The unsuccessful example showcases pronounced amplification bias and PCR conditions that are poorly optimized for the GC content of the majority of cCREs.<br></p>
</div>
<div id="activity-statistic-vs-count-ratio" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Activity statistic vs count ratio<a href="activity-qc.html#activity-statistic-vs-count-ratio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess the correlation between log2(RNA/DNA) and the summary statistic of activity <br>
<strong>Input file:</strong> comb_df <br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/Activity_statistic_vs_count_ratio.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L4a2/Activity_statistic_vs_count_ratio.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A scatter plot of raw actviity values x-axis vs the activity summary statistic. Colors show density.<br>
<strong>Interpretation:</strong> The successful example showcases an experiment with high complexity, reflected in the high correlation between the measurements. The unsuccessful example showcases an experiment with a relatively low correlation between measurements, suggesting low complexity<br></p>
</div>
<div id="activity-distribution" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Activity distribution<a href="activity-qc.html#activity-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess activity dynamic range, noise, and statistical power <br>
<strong>Input file:</strong> comb_df<br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity:<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:32%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/activity_distribution.svg" style="width:100%; display:block;"/>
</div>
<div style="width:32%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L4a2/activity_distribution.svg" style="width:100%; display:block;"/>
</div>
<div style="width:32%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/activity_distribution.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A histogram of log2(RNA/DNA) of the tested cCREs, with significantly active cCREs marked in red. <br>
<strong>Interpretation:</strong> The successful example showcases a broad dynamic range, with most of the highly active cCREs detected as significantly active. The 1st unsuccessful example showcases a problem in the detection of activity, either because of high noise levels or the experiment being underpowered. The 2nd unsuccessful example showcases silencing, with cCREs that are unlikely to be truly active detected as significantly active<br></p>
</div>
<div id="differential-activity-distribution" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Differential activity distribution<a href="activity-qc.html#differential-activity-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess differential activity dynamic range, noise, and statistical power <br>
<strong>Input file:</strong> comparative_df <br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/Differential_activity_distribution.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/Differential_activity_distribution.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A histogram of the log2(RNA/DNA) difference between the two alleles of a cCRE, with significantly differentially active cCREs marked in red. <br>
<strong>Interpretation:</strong> The successful example showcases a broad dynamic range of differential activity, with high fold-changes detected as significantly differential active. The unsuccessful example showcases underdetection of differential activity or high noise levels.<br></p>
</div>
<div id="p-value-distribution" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> P-value distribution<a href="activity-qc.html#p-value-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess statistical power <br>
<strong>Input file:</strong> comb_df<br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity, Reproducibility, Dynamic range<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/P_value_distribution.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/P_value_distribution.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A qq plot of the p-values assigned to each cCRE activity level compared to the expected p-values generated by a uniform distribution <br>
<strong>Interpretation:</strong> The successful example showcases sufficient statistical power, reflected in a downward deviation from the expected distribution, with many low p-values. The unsuccessful example showcases an upward deviation, reflecting an inflation of p-values.<br></p>
</div>
<div id="volcano-plot---fc-vs-pval" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Volcano plot - FC vs Pval<a href="activity-qc.html#volcano-plot---fc-vs-pval" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess statistical power of differential activity detection <br>
<strong>Input file:</strong> comparative_df<br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/Differential_activity_volcano_zoom.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/Differential_activity_volcano_zoom.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A volcano plot of logFC vs FDR. Dashed line shows FDR = 0.05. <br>
<strong>Interpretation:</strong> The successful example showcases sufficient statistical power to detect differential activity. cCRES with large absolute(logFC) are detected as differentially active. The unsuccessful example showcases many cCREs with relatively large absolute(logFC) that are not detected as differentially active, suggesting low power or high noise levels.<br></p>
</div>
<div id="bc-retention-by-dnarna-sequencing-depth" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> BC retention by DNA/RNA sequencing depth<a href="activity-qc.html#bc-retention-by-dnarna-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess whether sequencing depth is sufficient <br>
<strong>Input file:</strong> downsampling_ratio_path <br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/BC_retention_by_DNA_RNA_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/BC_retention_by_DNA_RNA_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> The relationship between sequencing depth and the percentage of retained BCs. Data is downsampled in order to assess whether we have reached saturation in the percentage of retained BCs. Then, data is extrapolated to predict whether additional sequencing would help improve the results. <br>
<strong>Interpretation:</strong> The successful example showcases sufficient sequencing depth, The baseline percentage is relatively high and additional sequencing doesn’t improve the percentage of retained BCs significantly. The unsuccessful example showcases an experiment with a low percentage of retained BCs and more sequencing increases significantly coverage suggesting additional sequencing is required<br></p>
</div>
<div id="ccre-retention-by-dnarna-sequencing-depth" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> cCRE retention by DNA/RNA sequencing depth<a href="activity-qc.html#ccre-retention-by-dnarna-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess whether sequencing depth is sufficient <br>
<strong>Input file:</strong> downsampling_ratio_path <br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/cCRE_retention_by_DNA_RNA_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/cCRE_retention_by_DNA_RNA_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A line plot that exhibits the effect of sequencing depth (x axis) on the percentage of retained cCREs (y axis). Sampling values higher than 1 represent predicted data points <br>
<strong>Interpretation:</strong> The successful example showcases sufficient sequencing depth, The baseline percentage is relatively high and additional sequencing doesn’t improve the percentage of retained cCREs significantly. The unsuccessful example showcases an experiment with a low percentage of retained cCREs and more sequencing increases significantly coverage suggesting additional sequencing is required<br></p>
</div>
<div id="activity-by-sequencing-depth" class="section level2 hasAnchor" number="3.10">
<h2><span class="header-section-number">3.10</span> Activity by sequencing depth<a href="activity-qc.html#activity-by-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess if sequencing depth is sufficient <br>
<strong>Input file:</strong> downsampling_activity_path<br>
<strong>Evaluated metrics:</strong> DNA Complexity, RNA Complexity<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/Activity_by_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/Activity_by_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> The relationship between sequencing depth and the percentage of active cCREs. Data is downsampled in order to assess whether we have reached saturation in the percentage of active cCREs. Then, data is extrapolated to predict whether additional sequencing would help improve the results. <br>
<strong>Interpretation:</strong> The successful example showcases sufficient sequencing depth: the percentage of active cCREs has reached a plateau, and more sequencing is not expected to affect it. The unsuccessful example showcases suboptimal RNA sequencing depth: additional sequencing will likely affect the percentage of active cCREs.<br></p>
</div>
<div id="cumulative-rna-reads" class="section level2 hasAnchor" number="3.11">
<h2><span class="header-section-number">3.11</span> Cumulative RNA reads<a href="activity-qc.html#cumulative-rna-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess jackpotting <br>
<strong>Input file:</strong> comb_df<br>
<strong>Evaluated metrics:</strong> RNA Complexity, Dynamic range<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/cumulative_RNA_reads.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/d2Osteoblast_spiking_oligos/cumulative_RNA_reads.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A cumulative distribution showcasing the fraction of RNA reads from each cCRE. cCREs are sorted by their RNA read counts. <br>
<strong>Interpretation:</strong> The successful example showcases no jackpotting – the RNA reads come from many cCREs. The unsuccessful example showcases an experiment where most of the RNA reads come from just a few cCREs.<br></p>
</div>
<div id="sample-clustering" class="section level2 hasAnchor" number="3.12">
<h2><span class="header-section-number">3.12</span> Sample clustering<a href="activity-qc.html#sample-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess reproducibility between samples <br>
<strong>Input file:</strong> cDNA_reads_by_cell_type<br>
<strong>Evaluated metrics:</strong> Reproducibility<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/PCA_similarity_between_samples.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/thylacine_biorxiv_Gallego_Romero/PCA_similarity_between_samples.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Principal component analysis (PCA) on the activity of cCREs in replicates from two cell lines. <br>
<strong>Interpretation:</strong> In the successful example, PC1 separates between cell types. In the unsuccessful example, variation between replicates is as large as that between cell types, and one of the replicates of cell type 2 clusters closely with cell type 1.<br></p>
</div>
<div id="correlation-between-replicates" class="section level2 hasAnchor" number="3.13">
<h2><span class="header-section-number">3.13</span> Correlation between replicates<a href="activity-qc.html#correlation-between-replicates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess reproducibility between replicates<br>
<strong>Input file:</strong> activity_per_rep<br>
<strong>Evaluated metrics:</strong> Reproducibility, Dynamic range<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/thylacine_biorxiv_Gallego_Romero/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L4a2/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Scatter plot of the log2(RNA/DNA) values of each cCRE across two experimental replicates. Each data point is a hexagon representing the local density of cCREs. <br>
<strong>Interpretation:</strong> The successful example showcases a high correlation between replicates in more active cCREs. The unsuccessful example showcases no correlation<br></p>
</div>
<div id="correlation-of-differential-activity-between-replicates" class="section level2 hasAnchor" number="3.14">
<h2><span class="header-section-number">3.14</span> Correlation of differential activity between replicates<a href="activity-qc.html#correlation-of-differential-activity-between-replicates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess reproducibility between replicates <br>
<strong>Input file:</strong> allelic_pairs_replicates_df<br>
<strong>Evaluated metrics:</strong> Reproducibility, Dynamic range <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/diff_activity_corr_reps_hexbin_w_bar.svg" width="49%" /><img src="external_figures/humanMPRA_L4a2/diff_activity_corr_reps_hexbin_w_bar.svg" width="49%" /></p>
<p><strong>Legend:</strong> Scatter plot of the log2(RNA/DNA) allelic difference values for each cCRE in two replicates. Each data point is a hexagon representing the local density of cCREs. <br>
<strong>Interpretation:</strong> The successful example showcases concordance in differential activity between repliactes. The unsuccessful example showcases little concordance.<br></p>
</div>
<div id="replicability-by-activity" class="section level2 hasAnchor" number="3.15">
<h2><span class="header-section-number">3.15</span> Replicability by activity<a href="activity-qc.html#replicability-by-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess the correlation in activity between replicates in active vs non-active cCREs <br>
<strong>Input file:</strong> activity_per_rep<br>
<strong>Evaluated metrics:</strong> Reproducibility <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/Variation_at_various_activity_levels.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/Variation_at_various_activity_levels.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Correlation between replicates as a function of average activity [log₂(RNA/DNA)], shown across fixed bins of width 1. <br>
<strong>Interpretation:</strong> In the successful example, more active cCREs showcase a higher correlation, which further improves with higher activity level. The unsuccessful example showcases a low correlation, which remains poor in active cCREs.<br></p>
</div>
<div id="correlation-between-replicates-controls" class="section level2 hasAnchor" number="3.16">
<h2><span class="header-section-number">3.16</span> Correlation between replicates (controls)<a href="activity-qc.html#correlation-between-replicates-controls" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess reproducibility between replicates in positive and negative controls <br>
<strong>Input file:</strong> comb_df<br>
<strong>Evaluated metrics:</strong> Reproducibility, Dynamic range<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Scatter plot of the log2(RNA/DNA) values of each cCRE across two experimental replicates. Positive controls are marked in green, and negative controls are marked in red. <br>
<strong>Interpretation:</strong> The successful example showcases a high correlation between positive controls and a low correlation between negative controls. The unsuccessful example showcases low correlations in both.<br></p>
</div>
<div id="cross-validation-allelic-pairs" class="section level2 hasAnchor" number="3.17">
<h2><span class="header-section-number">3.17</span> Cross-validation: allelic pairs<a href="activity-qc.html#cross-validation-allelic-pairs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess reproducibility between allelic pairs <br>
<strong>Input file:</strong> allelic_pairs_df <br>
<strong>Evaluated metrics:</strong> Reproducibility <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/allelic_pairs_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/allelic_pairs_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Log2(RNA/DNA) values of the two alleles of a cCRE. Each data point is a hexagon representing the local density of cCREs.<br>
<strong>Interpretation:</strong> When allelic pairs differ by only a few nucleotides, their regulatory activity should be highly correlated. The successful example showcases a strong correlation between allelic pairs. The unsuccessful example showcases little to no correlation.<br></p>
</div>
<div id="cross-validaiton-cell-types" class="section level2 hasAnchor" number="3.18">
<h2><span class="header-section-number">3.18</span> Cross-validaiton: cell types<a href="activity-qc.html#cross-validaiton-cell-types" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess reproducibility between different cell types<br>
<strong>Input file:</strong> cell_types_df <br>
<strong>Evaluated metrics:</strong> Reproducibility<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_NPC/cell_types_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/cell_types_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Log2(RNA/DNA) values of shared control sequences across two cell types. Each data point is a hexagon representing the local density of cCREs. <br>
<strong>Interpretation:</strong> Although cell types differ in their trans environments, many regulatory networks are shared, particularly between closely related cell types. As a result, regulatory sequences often exhibit similar activity across cell types. This pattern is evident in the successful example, which shows a strong correlation between cell types. The unsuccessful example, on the other hand, showcases no correlation.<br></p>
</div>
<div id="minimizing-noise" class="section level2 hasAnchor" number="3.19">
<h2><span class="header-section-number">3.19</span> Minimizing noise<a href="activity-qc.html#minimizing-noise" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Remove outlier barcodes and optimize minimum DNA count threshold to improve reproducibility<br>
<strong>Input file:</strong> different_std_threshold_analysis<br>
<strong>Evaluated metrics:</strong> Reproducibility<br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/minimizing_noise_hexbin.svg" width="100%" />
<strong>Legend:</strong> Each panel shows the correlation between replicates across different threshold of outlier removal (horizontal), and minimum DNA counts per cCRE (vertical). Each data point is a hexagon representing the local density of cCREs.<br>
<strong>Interpretation:</strong> The examples show an increase in correlations between replicates with increasing DNA cutoffs and outlier BC removal<br></p>
</div>
<div id="reproducibility-by-sequencing-depth" class="section level2 hasAnchor" number="3.20">
<h2><span class="header-section-number">3.20</span> Reproducibility by sequencing depth<a href="activity-qc.html#reproducibility-by-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess whether additional sequencing will improve reproducibility between replicates <br>
<strong>Input file:</strong> downsampling_activity_path, downsampling_ratio_path <br>
<strong>Evaluated metrics:</strong> Reproducibility <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/Reproducibility_by_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/Reproducibility_by_sequencing_depth.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> The correlation between replicates as a function of the sequencing depth. The x-axis shows increasing levels of downsampling of the current data. <br>
<strong>Interpretation:</strong> In the successful example, additional sequencing is expected to slightly improve reproducibility, whereas in the unsuccessful example it is not.<br></p>
</div>
<div id="rna-vs-dna" class="section level2 hasAnchor" number="3.21">
<h2><span class="header-section-number">3.21</span> RNA vs DNA<a href="activity-qc.html#rna-vs-dna" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess true activity in the experiment <br>
<strong>Input file:</strong> comb_df<br>
<strong>Evaluated metrics:</strong> Dynamic Range<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:32%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
<div style="width:32%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L4a2/RNA_DNA_ratio_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
<div style="width:32%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/RNA_DNA_ratio_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> RNA vs. DNA counts for each cCRE, normalized to sequencing depth. Each data point is a hexagon representing the local density of cCREs. <br>
<strong>Interpretation:</strong> The successful example showcases many sequences with high RNA-to-DNA ratios, reflected by points above the diagonal. The 1st unsuccessful example showcases little to no activity, with points clustering along the diagonal, and the 2nd unsuccessful example showcases silencing, reflected by many points falling below the diagonal.<br></p>
</div>
<div id="activity-of-controls" class="section level2 hasAnchor" number="3.22">
<h2><span class="header-section-number">3.22</span> Activity of controls<a href="activity-qc.html#activity-of-controls" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess the dynamic range of activity<br>
<strong>Input file:</strong> comb_df<br>
<strong>Evaluated metrics:</strong> Dynamic Range<br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/Control_activity_boxplots.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/Max_MPRA_run2/Control_activity_boxplots.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Box plots of activity levels for positive and negative controls, as well as cCREs. <br>
<strong>Interpretation:</strong> In the successful example, positive controls show high activity while negative controls show low activity. The unsuccessful example shows similar activity levels for positive and negative controls.<br></p>
</div>
<div id="genomic-annotations" class="section level2 hasAnchor" number="3.23">
<h2><span class="header-section-number">3.23</span> Genomic annotations<a href="activity-qc.html#genomic-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess concordance with endogenous signals of active chromatin marks<br>
<strong>Input file:</strong> screen_df <br>
<strong>Evaluated metrics:</strong> Concordance with endogenous signals <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/Genomic_annotations.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/Genomic_annotations.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> A stacked bar plot of cCRE overlap with active chromatin marks (ENCODE SCREEN elements). cCREs were grouped into six groups based on their activity levels. <br>
<strong>Interpretation:</strong> The successful example showcases increased overlap with active chromatin marks as activity increases. The unsuccessful example shows no such increase<br></p>
</div>
<div id="proximity-to-tss" class="section level2 hasAnchor" number="3.24">
<h2><span class="header-section-number">3.24</span> Proximity to TSS<a href="activity-qc.html#proximity-to-tss" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess concordance with endogenous locations of cCREs <br>
<strong>Input file:</strong> tss_df<br>
<strong>Evaluated metrics:</strong> Concordance with endogenous signals <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/PMID_38766054_Reilly/Proximity_to_TSS.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/Proximity_to_TSS.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Box plots of cCRE distance to the closest TSS. cCREs were grouped into six groups based on their activity levels. <br>
<strong>Interpretation:</strong> The successful example showcases that more active cCREs tend to be closer to a TSS. The unsuccessful example shows no such trend<br></p>
</div>
<div id="ai-predictions-vs-activity" class="section level2 hasAnchor" number="3.25">
<h2><span class="header-section-number">3.25</span> AI predictions vs activity<a href="activity-qc.html#ai-predictions-vs-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess the correlation between AI predictions of activity and MPRA results <br>
<strong>Input file:</strong> AI_df <br>
<strong>Evaluated metrics:</strong> Concordance with endogenous signals <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/AI_predictions_vs_activity_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_activity_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Experimentally measured activity vs AI-predicted activity. Each data point is a hexagon representing the local density of cCREs.<br>
<strong>Interpretation:</strong> The successful example showcases a high correlation, and the unsuccessful example showcases little correlation<br></p>
</div>
<div id="ai-predictions-vs-differential-activity" class="section level2 hasAnchor" number="3.26">
<h2><span class="header-section-number">3.26</span> AI predictions vs differential activity<a href="activity-qc.html#ai-predictions-vs-differential-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> Assess the correlation between AI predictions of differential activity and MPRA results <br>
<strong>Input file:</strong> AI_comparative_df <br>
<strong>Evaluated metrics:</strong> Concordance with endogenous signals <br></p>
<div style="display:flex; justify-content:space-between; align-items:flex-start; gap:16px; width:100%;">
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_up.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/modern_humanMPRA_Hob/AI_predictions_vs_differential_activity_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
<div style="width:49%;">
<div style="text-align:center; margin-bottom:6px;">
<img src="external_figures/thumbs_down.svg" style="width:60px; display:inline-block;"/>
</div>
<img src="external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_differential_activity_hexbin_w_bar.svg" style="width:100%; display:block;"/>
</div>
</div>
<p><strong>Legend:</strong> Experimentally measured differential activity vs AI-predicted differential activity. Each data point is a hexagon representing the local density of cCREs. <br>
<strong>Interpretation:</strong> The successful example showcases some correlation, and the unsuccessful example showcases no correlation.<br></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="associations-qc.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/12-only_figs_activity.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
