<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Activity QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Activity QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Activity QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Omer Ronen" />


<meta name="date" content="2026-01-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="associations-qc.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i>Usage</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#scripts"><i class="fa fa-check"></i>Scripts</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html"><i class="fa fa-check"></i><b>1</b> A guide for running the analyses notebook</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#associations"><i class="fa fa-check"></i><b>1.1</b> Associations</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#script"><i class="fa fa-check"></i><b>1.1.1</b> Script</a></li>
<li class="chapter" data-level="1.1.2" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#input"><i class="fa fa-check"></i><b>1.1.2</b> Input</a></li>
<li class="chapter" data-level="1.1.3" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#output"><i class="fa fa-check"></i><b>1.1.3</b> Output</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#activity"><i class="fa fa-check"></i><b>1.2</b> Activity</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#script-1"><i class="fa fa-check"></i><b>1.2.1</b> Script</a></li>
<li class="chapter" data-level="1.2.2" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#input-1"><i class="fa fa-check"></i><b>1.2.2</b> Input</a></li>
<li class="chapter" data-level="1.2.3" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#output-1"><i class="fa fa-check"></i><b>1.2.3</b> Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="associations-qc.html"><a href="associations-qc.html"><i class="fa fa-check"></i><b>2</b> Associations QC</a>
<ul>
<li class="chapter" data-level="2.1" data-path="associations-qc.html"><a href="associations-qc.html#bcs-per-ccre"><i class="fa fa-check"></i><b>2.1</b> BCs per cCRE</a></li>
<li class="chapter" data-level="2.2" data-path="associations-qc.html"><a href="associations-qc.html#pcr-bias---gc"><i class="fa fa-check"></i><b>2.2</b> PCR bias - GC</a></li>
<li class="chapter" data-level="2.3" data-path="associations-qc.html"><a href="associations-qc.html#pcr-bias---g-stretches"><i class="fa fa-check"></i><b>2.3</b> PCR bias - G Stretches</a></li>
<li class="chapter" data-level="2.4" data-path="associations-qc.html"><a href="associations-qc.html#umis-per-association"><i class="fa fa-check"></i><b>2.4</b> UMIs per association</a></li>
<li class="chapter" data-level="2.5" data-path="associations-qc.html"><a href="associations-qc.html#retained-ccres"><i class="fa fa-check"></i><b>2.5</b> Retained cCREs</a></li>
<li class="chapter" data-level="2.6" data-path="associations-qc.html"><a href="associations-qc.html#ccres-per-bc"><i class="fa fa-check"></i><b>2.6</b> cCREs per BC</a></li>
<li class="chapter" data-level="2.7" data-path="associations-qc.html"><a href="associations-qc.html#ccre-retention-by-sequencing-depth"><i class="fa fa-check"></i><b>2.7</b> cCRE retention by sequencing depth</a></li>
<li class="chapter" data-level="2.8" data-path="associations-qc.html"><a href="associations-qc.html#bcs-per-ccre-by-sequencing-depth"><i class="fa fa-check"></i><b>2.8</b> BCs per cCRE by sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="activity-qc.html"><a href="activity-qc.html"><i class="fa fa-check"></i><b>3</b> Activity QC</a>
<ul>
<li class="chapter" data-level="3.1" data-path="activity-qc.html"><a href="activity-qc.html#retained-ccres-and-barcodes"><i class="fa fa-check"></i><b>3.1</b> Retained cCREs and barcodes</a></li>
<li class="chapter" data-level="3.2" data-path="activity-qc.html"><a href="activity-qc.html#dna-counts-vs-gc-content"><i class="fa fa-check"></i><b>3.2</b> DNA counts vs GC content</a></li>
<li class="chapter" data-level="3.3" data-path="activity-qc.html"><a href="activity-qc.html#activity-statistic-vs-count-ratio"><i class="fa fa-check"></i><b>3.3</b> Activity statistic vs count ratio</a></li>
<li class="chapter" data-level="3.4" data-path="activity-qc.html"><a href="activity-qc.html#differential-activity-statistic-vs-count-ratio"><i class="fa fa-check"></i><b>3.4</b> Differential activity statistic vs count ratio</a></li>
<li class="chapter" data-level="3.5" data-path="activity-qc.html"><a href="activity-qc.html#activity-distribution"><i class="fa fa-check"></i><b>3.5</b> Activity distribution</a></li>
<li class="chapter" data-level="3.6" data-path="activity-qc.html"><a href="activity-qc.html#differential-activity-distribution"><i class="fa fa-check"></i><b>3.6</b> Differential activity distribution</a></li>
<li class="chapter" data-level="3.7" data-path="activity-qc.html"><a href="activity-qc.html#p-value-distribution"><i class="fa fa-check"></i><b>3.7</b> P-value distribution</a></li>
<li class="chapter" data-level="3.8" data-path="activity-qc.html"><a href="activity-qc.html#volcano-plot---fc-vs-pval"><i class="fa fa-check"></i><b>3.8</b> Volcano plot - FC vs Pval</a></li>
<li class="chapter" data-level="3.9" data-path="activity-qc.html"><a href="activity-qc.html#bc-retention-by-dnarna-sequencing-depth"><i class="fa fa-check"></i><b>3.9</b> BC retention by DNA/RNA sequencing depth</a></li>
<li class="chapter" data-level="3.10" data-path="activity-qc.html"><a href="activity-qc.html#ccre-retention-by-dnarna-sequencing-depth"><i class="fa fa-check"></i><b>3.10</b> cCRE retention by DNA/RNA sequencing depth</a></li>
<li class="chapter" data-level="3.11" data-path="activity-qc.html"><a href="activity-qc.html#activity-by-sequencing-depth"><i class="fa fa-check"></i><b>3.11</b> Activity by sequencing depth</a></li>
<li class="chapter" data-level="3.12" data-path="activity-qc.html"><a href="activity-qc.html#cumulative-rna-reads"><i class="fa fa-check"></i><b>3.12</b> Cumulative RNA reads</a></li>
<li class="chapter" data-level="3.13" data-path="activity-qc.html"><a href="activity-qc.html#sample-clustering"><i class="fa fa-check"></i><b>3.13</b> Sample clustering</a></li>
<li class="chapter" data-level="3.14" data-path="activity-qc.html"><a href="activity-qc.html#correlation-between-replicates"><i class="fa fa-check"></i><b>3.14</b> Correlation between replicates</a></li>
<li class="chapter" data-level="3.15" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-replicates"><i class="fa fa-check"></i><b>3.15</b> Correlation of differential activity between replicates</a></li>
<li class="chapter" data-level="3.16" data-path="activity-qc.html"><a href="activity-qc.html#replicability-by-activity"><i class="fa fa-check"></i><b>3.16</b> Replicability by activity</a></li>
<li class="chapter" data-level="3.17" data-path="activity-qc.html"><a href="activity-qc.html#correlation-between-replicates-controls"><i class="fa fa-check"></i><b>3.17</b> Correlation between replicates (controls)</a></li>
<li class="chapter" data-level="3.18" data-path="activity-qc.html"><a href="activity-qc.html#cross-validationallelic-pairs"><i class="fa fa-check"></i><b>3.18</b> Cross-validation:allelic pairs</a></li>
<li class="chapter" data-level="3.19" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-overlapping-sequences"><i class="fa fa-check"></i><b>3.19</b> Correlation of differential activity between overlapping sequences</a></li>
<li class="chapter" data-level="3.20" data-path="activity-qc.html"><a href="activity-qc.html#cross-validaiton-cell-types"><i class="fa fa-check"></i><b>3.20</b> Cross-validaiton: cell types</a></li>
<li class="chapter" data-level="3.21" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-cell-types"><i class="fa fa-check"></i><b>3.21</b> Correlation of differential activity between cell types</a></li>
<li class="chapter" data-level="3.22" data-path="activity-qc.html"><a href="activity-qc.html#cross-validation-experiments"><i class="fa fa-check"></i><b>3.22</b> Cross-validation: experiments</a></li>
<li class="chapter" data-level="3.23" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-experiments"><i class="fa fa-check"></i><b>3.23</b> Correlation of differential activity between experiments</a></li>
<li class="chapter" data-level="3.24" data-path="activity-qc.html"><a href="activity-qc.html#replicability-across-experiments"><i class="fa fa-check"></i><b>3.24</b> Replicability across experiments</a></li>
<li class="chapter" data-level="3.25" data-path="activity-qc.html"><a href="activity-qc.html#minimizing-noise"><i class="fa fa-check"></i><b>3.25</b> Minimizing noise</a></li>
<li class="chapter" data-level="3.26" data-path="activity-qc.html"><a href="activity-qc.html#reproducibility-by-sequencing-depth"><i class="fa fa-check"></i><b>3.26</b> Reproducibility by sequencing depth</a></li>
<li class="chapter" data-level="3.27" data-path="activity-qc.html"><a href="activity-qc.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.27</b> RNA vs DNA</a></li>
<li class="chapter" data-level="3.28" data-path="activity-qc.html"><a href="activity-qc.html#activity-of-controls"><i class="fa fa-check"></i><b>3.28</b> Activity of controls</a></li>
<li class="chapter" data-level="3.29" data-path="activity-qc.html"><a href="activity-qc.html#genomic-annotations"><i class="fa fa-check"></i><b>3.29</b> Genomic annotations</a></li>
<li class="chapter" data-level="3.30" data-path="activity-qc.html"><a href="activity-qc.html#proximity-to-tss"><i class="fa fa-check"></i><b>3.30</b> Proximity to TSS</a></li>
<li class="chapter" data-level="3.31" data-path="activity-qc.html"><a href="activity-qc.html#ai-predictions-vs-activity"><i class="fa fa-check"></i><b>3.31</b> AI predictions vs activity</a></li>
<li class="chapter" data-level="3.32" data-path="activity-qc.html"><a href="activity-qc.html#ai-predictions-vs-differential-activity"><i class="fa fa-check"></i><b>3.32</b> AI predictions vs differential activity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="activity-qc" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Activity QC<a href="activity-qc.html#activity-qc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="retained-ccres-and-barcodes" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Retained cCREs and barcodes<a href="activity-qc.html#retained-ccres-and-barcodes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Assesses cCRE and BC coverage</strong> <br>
<strong>Input file: activity_per_rep</strong><br>
<strong>Evaluated metrics: DNA Complexity, RNA Complexity</strong>:<br></p>
<p><img src="external_figures/PMID_38766054_Reilly/retained_cCREs_and_barcodes.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/retained_cCREs_and_barcodes.svg" width="49%" /></p>
<p><strong>Legend: Percentage of cCREs and BCs present in the DNA and RNA quantification data compared with those observed in the association data</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="dna-counts-vs-gc-content" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> DNA counts vs GC content<a href="activity-qc.html#dna-counts-vs-gc-content" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/PMID_38766054_Reilly/GC_content_bias_in_DNA.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/GC_content_bias_in_DNA.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="activity-statistic-vs-count-ratio" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Activity statistic vs count ratio<a href="activity-qc.html#activity-statistic-vs-count-ratio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/PMID_38766054_Reilly/Activity_statistic_vs_count_ratio.svg" width="49%" /><img src="external_figures/humanMPRA_L4a2/Activity_statistic_vs_count_ratio.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="differential-activity-statistic-vs-count-ratio" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Differential activity statistic vs count ratio<a href="activity-qc.html#differential-activity-statistic-vs-count-ratio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="activity-distribution" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Activity distribution<a href="activity-qc.html#activity-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Assesses activity dynamic range, noise, and statistical power</strong> <br>
<strong>Input file: quantification table (comb_df)</strong><br>
<strong>Evaluated metrics: DNA Complexity, RNA Complexity</strong>:<br></p>
<p><img src="external_figures/PMID_38766054_Reilly/activity_distribution.svg" width="32%" /><img src="external_figures/humanMPRA_L4a2/activity_distribution.svg" width="32%" /><img src="external_figures/humanMPRA_L1a1_Neurons/activity_distribution.svg" width="32%" />
<strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="differential-activity-distribution" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Differential activity distribution<a href="activity-qc.html#differential-activity-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Assesses differential activity dynamic range, noise, and statistical power</strong> <br>
<strong>Input file: comparative_df</strong> <br>
<strong>Evaluated metrics: DNA Complexity, RNA Complexity</strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/Differential_activity_distribution.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/Differential_activity_distribution.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="p-value-distribution" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> P-value distribution<a href="activity-qc.html#p-value-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Inferring the power of the statistical test</strong> <br>
<strong>Input file: quantification table (comb_df)</strong><br>
<strong>Evaluated metrics: DNA Complexity, RNA Complexity, Reproducibility, Dynamic range</strong>:<br></p>
<p><img src="external_figures/PMID_38766054_Reilly/P_value_distribution.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/P_value_distribution.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="volcano-plot---fc-vs-pval" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Volcano plot - FC vs Pval<a href="activity-qc.html#volcano-plot---fc-vs-pval" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/Differential_activity_volcano_zoom.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/Differential_activity_volcano_zoom.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="bc-retention-by-dnarna-sequencing-depth" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> BC retention by DNA/RNA sequencing depth<a href="activity-qc.html#bc-retention-by-dnarna-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/BC_retention_by_DNA_RNA_sequencing_depth.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/BC_retention_by_DNA_RNA_sequencing_depth.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="ccre-retention-by-dnarna-sequencing-depth" class="section level2 hasAnchor" number="3.10">
<h2><span class="header-section-number">3.10</span> cCRE retention by DNA/RNA sequencing depth<a href="activity-qc.html#ccre-retention-by-dnarna-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/cCRE_retention_by_DNA_RNA_sequencing_depth.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/cCRE_retention_by_DNA_RNA_sequencing_depth.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="activity-by-sequencing-depth" class="section level2 hasAnchor" number="3.11">
<h2><span class="header-section-number">3.11</span> Activity by sequencing depth<a href="activity-qc.html#activity-by-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: uses downsampling of sequencing reads to assess if sequencing depth is sufficient</strong> <br>
<strong>Input file: Downsampling activity data frames</strong><br>
<strong>Evaluated metrics: DNA Complexity, RNA Complexity</strong><br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/Activity_by_sequencing_depth.svg" width="49%" /><img src="external_figures/modern_humanMPRA_NPC/Activity_by_sequencing_depth.svg" width="49%" />
<strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="cumulative-rna-reads" class="section level2 hasAnchor" number="3.12">
<h2><span class="header-section-number">3.12</span> Cumulative RNA reads<a href="activity-qc.html#cumulative-rna-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: assesses jackpotting</strong> <br>
<strong>Input file: quantification table (comb_df)</strong><br>
<strong>Evaluated metrics: DNA Complexity, RNA Complexity</strong>:<br></p>
<p><img src="external_figures/PMID_38766054_Reilly/cumulative_RNA_reads.svg" width="49%" /><img src="external_figures/d2Osteoblast_spiking_oligos/cumulative_RNA_reads.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="sample-clustering" class="section level2 hasAnchor" number="3.13">
<h2><span class="header-section-number">3.13</span> Sample clustering<a href="activity-qc.html#sample-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Assesses reproducibility between samples</strong> <br>
<strong>Input file: cDNA_reads_by_cell_type</strong><br>
<strong>Evaluated metrics: Reproducibility</strong>:<br></p>
<p>mention in the bookdown: the importance of the percentage explained by the 1st and 2nd PCs.</p>
<p><img src="external_figures/PMID_38766054_Reilly/PCA_similarity_between_samples.svg" width="49%" /><img src="external_figures/thylacine_biorxiv_Gallego_Romero/PCA_similarity_between_samples.svg" width="49%" /></p>
<p><strong>Legend: Principal component analysis (PCA) on samples</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="correlation-between-replicates" class="section level2 hasAnchor" number="3.14">
<h2><span class="header-section-number">3.14</span> Correlation between replicates<a href="activity-qc.html#correlation-between-replicates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: assesses reproducibility between replicates</strong> <br>
<strong>Input file: activity_per_rep</strong><br>
<strong>Evaluated metrics: Reproducibility</strong>:<br></p>
<p><img src="external_figures/thylacine_biorxiv_Gallego_Romero/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg" width="49%" /><img src="external_figures/humanMPRA_L4a2/RNA_DNA_ratio_correlation_between_replicates_hexbin_w_bar.svg" width="49%" />
<strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="correlation-of-differential-activity-between-replicates" class="section level2 hasAnchor" number="3.15">
<h2><span class="header-section-number">3.15</span> Correlation of differential activity between replicates<a href="activity-qc.html#correlation-of-differential-activity-between-replicates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="replicability-by-activity" class="section level2 hasAnchor" number="3.16">
<h2><span class="header-section-number">3.16</span> Replicability by activity<a href="activity-qc.html#replicability-by-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: assesses the correlation in activity between replicates in active vs non-active cCREs</strong> <br>
<strong>Input file:</strong><br>
<strong>Evaluated metrics: Reproducibility </strong><br></p>
<p><img src="external_figures/PMID_38766054_Reilly/Variation_at_various_activity_levels.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/Variation_at_various_activity_levels.svg" width="49%" />
<strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="correlation-between-replicates-controls" class="section level2 hasAnchor" number="3.17">
<h2><span class="header-section-number">3.17</span> Correlation between replicates (controls)<a href="activity-qc.html#correlation-between-replicates-controls" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: analysis assesses reproducibility between replicates in positive and negative controls</strong> <br>
<strong>Input file: quantification table (comb_df)</strong><br>
<strong>Evaluated metrics: Reproducibility</strong><br></p>
<p><img src="external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/RNA_DNA_ratio_correlation_between_replicates_with_controls_hexbin.svg" width="49%" />
<strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="cross-validationallelic-pairs" class="section level2 hasAnchor" number="3.18">
<h2><span class="header-section-number">3.18</span> Cross-validation:allelic pairs<a href="activity-qc.html#cross-validationallelic-pairs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/allelic_pairs_hexbin_w_bar.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/allelic_pairs_hexbin_w_bar.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="correlation-of-differential-activity-between-overlapping-sequences" class="section level2 hasAnchor" number="3.19">
<h2><span class="header-section-number">3.19</span> Correlation of differential activity between overlapping sequences<a href="activity-qc.html#correlation-of-differential-activity-between-overlapping-sequences" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="cross-validaiton-cell-types" class="section level2 hasAnchor" number="3.20">
<h2><span class="header-section-number">3.20</span> Cross-validaiton: cell types<a href="activity-qc.html#cross-validaiton-cell-types" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="correlation-of-differential-activity-between-cell-types" class="section level2 hasAnchor" number="3.21">
<h2><span class="header-section-number">3.21</span> Correlation of differential activity between cell types<a href="activity-qc.html#correlation-of-differential-activity-between-cell-types" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="cross-validation-experiments" class="section level2 hasAnchor" number="3.22">
<h2><span class="header-section-number">3.22</span> Cross-validation: experiments<a href="activity-qc.html#cross-validation-experiments" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="correlation-of-differential-activity-between-experiments" class="section level2 hasAnchor" number="3.23">
<h2><span class="header-section-number">3.23</span> Correlation of differential activity between experiments<a href="activity-qc.html#correlation-of-differential-activity-between-experiments" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="replicability-across-experiments" class="section level2 hasAnchor" number="3.24">
<h2><span class="header-section-number">3.24</span> Replicability across experiments<a href="activity-qc.html#replicability-across-experiments" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/placeholder.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="minimizing-noise" class="section level2 hasAnchor" number="3.25">
<h2><span class="header-section-number">3.25</span> Minimizing noise<a href="activity-qc.html#minimizing-noise" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Removing outlier BCs and optimizing the threshold of minimum DNA counts to increase reproducibility</strong> <br>
<strong>Input file: </strong><br>
<strong>Evaluated metrics: Reproducibility</strong>:<br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/minimizing_noise_hexbin.svg" width="100%" />
<strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="reproducibility-by-sequencing-depth" class="section level2 hasAnchor" number="3.26">
<h2><span class="header-section-number">3.26</span> Reproducibility by sequencing depth<a href="activity-qc.html#reproducibility-by-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Assesses whether additional sequencing will improve reproducibility between replicates</strong> <br>
<strong>Input file: Activity down sampling files</strong> <br>
<strong>Evaluated metrics: Reproducibility</strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/Reproducibility_by_sequencing_depth.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/Reproducibility_by_sequencing_depth.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="rna-vs-dna" class="section level2 hasAnchor" number="3.27">
<h2><span class="header-section-number">3.27</span> RNA vs DNA<a href="activity-qc.html#rna-vs-dna" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Assesses if there is true activity in the experiment</strong> <br>
<strong>Input file:</strong><br>
<strong>Evaluated metrics: Dynamic Range</strong><br></p>
<p><img src="external_figures/PMID_38766054_Reilly/RNA_DNA_ratio_hexbin_w_bar.svg" width="32%" /><img src="external_figures/humanMPRA_L4a2/RNA_DNA_ratio_hexbin_w_bar.svg" width="32%" /><img src="external_figures/humanMPRA_L1a1_Neurons/RNA_DNA_ratio_hexbin_w_bar.svg" width="32%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="activity-of-controls" class="section level2 hasAnchor" number="3.28">
<h2><span class="header-section-number">3.28</span> Activity of controls<a href="activity-qc.html#activity-of-controls" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: assesses the dynamic range of activity</strong> <br>
<strong>Input file: quantification table (comb_df)</strong><br>
<strong>Evaluated metrics: Dynamic Range</strong>:<br></p>
<p><img src="external_figures/PMID_38766054_Reilly/Control_activity_boxplots.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/Control_activity_boxplots.svg" width="49%" />
<strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="genomic-annotations" class="section level2 hasAnchor" number="3.29">
<h2><span class="header-section-number">3.29</span> Genomic annotations<a href="activity-qc.html#genomic-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: This analysis assesses concordance with endogenous signals of active chromatin marks</strong> <br>
<strong>Input file: Genomic annotation file</strong> <br>
<strong>Evaluated metrics: Dynamic Range</strong> <br></p>
<p><img src="external_figures/PMID_38766054_Reilly/Genomic_annotations.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/Genomic_annotations.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="proximity-to-tss" class="section level2 hasAnchor" number="3.30">
<h2><span class="header-section-number">3.30</span> Proximity to TSS<a href="activity-qc.html#proximity-to-tss" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: is analysis assesses concordance with endogenous locations of cCREs</strong> <br>
<strong>Input file: Distance to TSS file</strong> <br>
<strong>Evaluated metrics: Dynamic Range</strong> <br></p>
<p><img src="external_figures/PMID_38766054_Reilly/Proximity_to_TSS.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/Proximity_to_TSS.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="ai-predictions-vs-activity" class="section level2 hasAnchor" number="3.31">
<h2><span class="header-section-number">3.31</span> AI predictions vs activity<a href="activity-qc.html#ai-predictions-vs-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/AI_predictions_vs_activity_hexbin_w_bar.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_activity_hexbin_w_bar.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="ai-predictions-vs-differential-activity" class="section level2 hasAnchor" number="3.32">
<h2><span class="header-section-number">3.32</span> AI predictions vs differential activity<a href="activity-qc.html#ai-predictions-vs-differential-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: </strong> <br>
<strong>Evaluated metrics: </strong> <br></p>
<p><img src="external_figures/modern_humanMPRA_Hob/AI_predictions_vs_differential_activity_hexbin_w_bar.svg" width="49%" /><img src="external_figures/humanMPRA_L1a1_Neurons/AI_predictions_vs_differential_activity_hexbin_w_bar.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="associations-qc.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/12-only_figs_activity.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
