<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Associations QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Associations QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Associations QC | Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Omer Ronen" />


<meta name="date" content="2026-01-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="a-guide-for-running-the-analyses-notebook.html"/>
<link rel="next" href="activity-qc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i>Usage</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#scripts"><i class="fa fa-check"></i>Scripts</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html"><i class="fa fa-check"></i><b>1</b> A guide for running the analyses notebook</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#associations"><i class="fa fa-check"></i><b>1.1</b> Associations</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#input"><i class="fa fa-check"></i><b>1.1.1</b> Input</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#activity"><i class="fa fa-check"></i><b>1.2</b> Activity</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#input-1"><i class="fa fa-check"></i><b>1.2.1</b> Input</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#differential-activity"><i class="fa fa-check"></i><b>1.3</b> Differential Activity</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="a-guide-for-running-the-analyses-notebook.html"><a href="a-guide-for-running-the-analyses-notebook.html#input-2"><i class="fa fa-check"></i><b>1.3.1</b> Input</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="associations-qc.html"><a href="associations-qc.html"><i class="fa fa-check"></i><b>2</b> Associations QC</a>
<ul>
<li class="chapter" data-level="2.1" data-path="associations-qc.html"><a href="associations-qc.html#bcs-per-ccre"><i class="fa fa-check"></i><b>2.1</b> BCs per cCRE</a></li>
<li class="chapter" data-level="2.2" data-path="associations-qc.html"><a href="associations-qc.html#pcr-bias---gc"><i class="fa fa-check"></i><b>2.2</b> PCR bias - GC</a></li>
<li class="chapter" data-level="2.3" data-path="associations-qc.html"><a href="associations-qc.html#pcr-bias---g-stretches"><i class="fa fa-check"></i><b>2.3</b> PCR bias - G Stretches</a></li>
<li class="chapter" data-level="2.4" data-path="associations-qc.html"><a href="associations-qc.html#umis-per-association"><i class="fa fa-check"></i><b>2.4</b> UMIs per association</a></li>
<li class="chapter" data-level="2.5" data-path="associations-qc.html"><a href="associations-qc.html#retained-ccres"><i class="fa fa-check"></i><b>2.5</b> Retained cCREs</a></li>
<li class="chapter" data-level="2.6" data-path="associations-qc.html"><a href="associations-qc.html#ccres-per-bc"><i class="fa fa-check"></i><b>2.6</b> cCREs per BC</a></li>
<li class="chapter" data-level="2.7" data-path="associations-qc.html"><a href="associations-qc.html#ccre-retention-by-sequencing-depth"><i class="fa fa-check"></i><b>2.7</b> cCRE retention by sequencing depth</a></li>
<li class="chapter" data-level="2.8" data-path="associations-qc.html"><a href="associations-qc.html#bcs-per-ccre-by-sequencing-depth"><i class="fa fa-check"></i><b>2.8</b> BCs per cCRE by sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="activity-qc.html"><a href="activity-qc.html"><i class="fa fa-check"></i><b>3</b> Activity QC</a>
<ul>
<li class="chapter" data-level="3.1" data-path="activity-qc.html"><a href="activity-qc.html#retained-ccres-and-barcodes"><i class="fa fa-check"></i><b>3.1</b> Retained cCREs and barcodes</a></li>
<li class="chapter" data-level="3.2" data-path="activity-qc.html"><a href="activity-qc.html#dna-counts-vs-gc-content"><i class="fa fa-check"></i><b>3.2</b> DNA counts vs GC content</a></li>
<li class="chapter" data-level="3.3" data-path="activity-qc.html"><a href="activity-qc.html#activity-statistic-vs-count-ratio"><i class="fa fa-check"></i><b>3.3</b> Activity statistic vs count ratio</a></li>
<li class="chapter" data-level="3.4" data-path="activity-qc.html"><a href="activity-qc.html#differential-activity-statistic-vs-count-ratio"><i class="fa fa-check"></i><b>3.4</b> Differential activity statistic vs count ratio</a></li>
<li class="chapter" data-level="3.5" data-path="activity-qc.html"><a href="activity-qc.html#activity-distribution"><i class="fa fa-check"></i><b>3.5</b> Activity distribution</a></li>
<li class="chapter" data-level="3.6" data-path="activity-qc.html"><a href="activity-qc.html#differential-activity-distribution"><i class="fa fa-check"></i><b>3.6</b> Differential activity distribution</a></li>
<li class="chapter" data-level="3.7" data-path="activity-qc.html"><a href="activity-qc.html#p-value-distribution"><i class="fa fa-check"></i><b>3.7</b> P-value distribution</a></li>
<li class="chapter" data-level="3.8" data-path="activity-qc.html"><a href="activity-qc.html#volcano-plot---fc-vs-pval"><i class="fa fa-check"></i><b>3.8</b> Volcano plot - FC vs Pval</a></li>
<li class="chapter" data-level="3.9" data-path="activity-qc.html"><a href="activity-qc.html#bc-retention-by-dnarna-sequencing-depth"><i class="fa fa-check"></i><b>3.9</b> BC retention by DNA/RNA sequencing depth</a></li>
<li class="chapter" data-level="3.10" data-path="activity-qc.html"><a href="activity-qc.html#ccre-retention-by-dnarna-sequencing-depth"><i class="fa fa-check"></i><b>3.10</b> cCRE retention by DNA/RNA sequencing depth</a></li>
<li class="chapter" data-level="3.11" data-path="activity-qc.html"><a href="activity-qc.html#activity-by-sequencing-depth"><i class="fa fa-check"></i><b>3.11</b> Activity by sequencing depth</a></li>
<li class="chapter" data-level="3.12" data-path="activity-qc.html"><a href="activity-qc.html#cumulative-rna-reads"><i class="fa fa-check"></i><b>3.12</b> Cumulative RNA reads</a></li>
<li class="chapter" data-level="3.13" data-path="activity-qc.html"><a href="activity-qc.html#sample-clustering"><i class="fa fa-check"></i><b>3.13</b> Sample clustering</a></li>
<li class="chapter" data-level="3.14" data-path="activity-qc.html"><a href="activity-qc.html#correlation-between-replicates"><i class="fa fa-check"></i><b>3.14</b> Correlation between replicates</a></li>
<li class="chapter" data-level="3.15" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-replicates"><i class="fa fa-check"></i><b>3.15</b> Correlation of differential activity between replicates</a></li>
<li class="chapter" data-level="3.16" data-path="activity-qc.html"><a href="activity-qc.html#replicability-by-activity"><i class="fa fa-check"></i><b>3.16</b> Replicability by activity</a></li>
<li class="chapter" data-level="3.17" data-path="activity-qc.html"><a href="activity-qc.html#correlation-between-replicates-controls"><i class="fa fa-check"></i><b>3.17</b> Correlation between replicates (controls)</a></li>
<li class="chapter" data-level="3.18" data-path="activity-qc.html"><a href="activity-qc.html#cross-validationallelic-pairs"><i class="fa fa-check"></i><b>3.18</b> Cross-validation:allelic pairs</a></li>
<li class="chapter" data-level="3.19" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-overlapping-sequences"><i class="fa fa-check"></i><b>3.19</b> Correlation of differential activity between overlapping sequences</a></li>
<li class="chapter" data-level="3.20" data-path="activity-qc.html"><a href="activity-qc.html#cross-validaiton-cell-types"><i class="fa fa-check"></i><b>3.20</b> Cross-validaiton: cell types</a></li>
<li class="chapter" data-level="3.21" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-cell-types"><i class="fa fa-check"></i><b>3.21</b> Correlation of differential activity between cell types</a></li>
<li class="chapter" data-level="3.22" data-path="activity-qc.html"><a href="activity-qc.html#cross-validation-experiments"><i class="fa fa-check"></i><b>3.22</b> Cross-validation: experiments</a></li>
<li class="chapter" data-level="3.23" data-path="activity-qc.html"><a href="activity-qc.html#correlation-of-differential-activity-between-experiments"><i class="fa fa-check"></i><b>3.23</b> Correlation of differential activity between experiments</a></li>
<li class="chapter" data-level="3.24" data-path="activity-qc.html"><a href="activity-qc.html#replicability-across-experiments"><i class="fa fa-check"></i><b>3.24</b> Replicability across experiments</a></li>
<li class="chapter" data-level="3.25" data-path="activity-qc.html"><a href="activity-qc.html#minimizing-noise"><i class="fa fa-check"></i><b>3.25</b> Minimizing noise</a></li>
<li class="chapter" data-level="3.26" data-path="activity-qc.html"><a href="activity-qc.html#reproducibility-by-sequencing-depth"><i class="fa fa-check"></i><b>3.26</b> Reproducibility by sequencing depth</a></li>
<li class="chapter" data-level="3.27" data-path="activity-qc.html"><a href="activity-qc.html#rna-vs-dna"><i class="fa fa-check"></i><b>3.27</b> RNA vs DNA</a></li>
<li class="chapter" data-level="3.28" data-path="activity-qc.html"><a href="activity-qc.html#activity-of-controls"><i class="fa fa-check"></i><b>3.28</b> Activity of controls</a></li>
<li class="chapter" data-level="3.29" data-path="activity-qc.html"><a href="activity-qc.html#genomic-annotations"><i class="fa fa-check"></i><b>3.29</b> Genomic annotations</a></li>
<li class="chapter" data-level="3.30" data-path="activity-qc.html"><a href="activity-qc.html#ai-predictions-vs-differential-activity"><i class="fa fa-check"></i><b>3.30</b> AI predictions vs differential activity</a></li>
<li class="chapter" data-level="3.31" data-path="activity-qc.html"><a href="activity-qc.html#proximity-to-tss"><i class="fa fa-check"></i><b>3.31</b> Proximity to TSS</a></li>
<li class="chapter" data-level="3.32" data-path="activity-qc.html"><a href="activity-qc.html#ai-predictions-vs-activity"><i class="fa fa-check"></i><b>3.32</b> AI predictions vs activity</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quality Control Pipeline for Massively Parallel Reporter Assays (MPRAs)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="associations-qc" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Associations QC<a href="associations-qc.html#associations-qc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="bcs-per-ccre" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> BCs per cCRE<a href="associations-qc.html#bcs-per-ccre" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: An empirical cumulative function (eCDF) of BCs per cCRE</strong> <br>
<strong>Input file: BC-cCRE association file</strong> <br>
<strong>Evaluated metrics: BC replicability</strong> <br></p>
<p><img src="external_figures/Max_MPRA_run2/Barcodes_per_cCRE.svg" width="49%" /><img src="external_figures/PMID_38766054_Reilly/Barcodes_per_cCRE.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="pcr-bias---gc" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> PCR bias - GC<a href="associations-qc.html#pcr-bias---gc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: This analysis assesses GC content bias in PCR amplification</strong> <br>
<strong>Input file: BC-cCRE association file</strong><br>
<strong>Evaluated metrics: Complexity</strong><br>
<img src="external_figures/Max_MPRA_run2/PCR_bias_GC.svg" width="49%" /><img src="external_figures/PMID_38766054_Reilly/PCR_bias_GC.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation: The successful example shows a relatively consistent number of reads per GC content, and PCR conditions that are optimized for the GC content levels of most cCREs (peaks are close to one another). The unsuccessful example shows both a strong amplification bias and suboptimal PCR conditions for the GC content levels of most cCREs</strong><br></p>
</div>
<div id="pcr-bias---g-stretches" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> PCR bias - G Stretches<a href="associations-qc.html#pcr-bias---g-stretches" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: This analysis assesses G-stretches bias in PCR amplification</strong> <br>
<strong>Input file: BC-cCRE association file</strong><br>
<strong>Evaluated metrics: Complexity</strong><br></p>
<p><img src="external_figures/placeholder.svg" width="49%" /><img src="external_figures/thylacine_biorxiv_Gallego_Romero/PCR_bias_G-stretches.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="umis-per-association" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> UMIs per association<a href="associations-qc.html#umis-per-association" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> <br>
<strong>Input file: Associations file before filtering for minimal number of associations</strong><br>
<strong>Evaluated metrics</strong>:<br>
<img src="external_figures/Max_MPRA_run2/UMIs_per_association.svg" width="49%" /><img src="external_figures/PMID_38766054_Reilly/UMIs_per_association.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="retained-ccres" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Retained cCREs<a href="associations-qc.html#retained-ccres" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: Retained cCREs per increasing cutoffs of BC number per cCRE</strong> <br>
<strong>Input file: BC-cCRE association file</strong><br>
<strong>Evaluated metrics: BC replicability</strong><br>
<img src="external_figures/Max_MPRA_run2/Retained_cCREs.svg" width="49%" /><img src="external_figures/PMID_38766054_Reilly/Retained_cCREs.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="ccres-per-bc" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> cCREs per BC<a href="associations-qc.html#ccres-per-bc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal:</strong> <br>
<strong>Input file: Associations file before filtering for BC promiscuity</strong><br>
<strong>Evaluated metrics: BC promiscuity</strong><br>
<img src="external_figures/Max_MPRA_run2/cCREs_per_BC_shuffled.svg" width="49%" /><img src="external_figures/PMID_38766054_Reilly/cCREs_per_BC.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="ccre-retention-by-sequencing-depth" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> cCRE retention by sequencing depth<a href="associations-qc.html#ccre-retention-by-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: This analysis uses downsampling of sequencing reads to assess whether sequencing depth is sufficient</strong> <br>
<strong>Input file: Downsampling association files</strong><br>
<strong>Evaluated metrics: complexity</strong><br>
<img src="external_figures/Max_MPRA_run2/Downsampling_Retained_cCREs.svg" width="49%" /><img src="external_figures/L3a2_281122/Downsampling_Retained_cCREs.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>
</div>
<div id="bcs-per-ccre-by-sequencing-depth" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> BCs per cCRE by sequencing depth<a href="associations-qc.html#bcs-per-ccre-by-sequencing-depth" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Goal: </strong> <br>
<strong>Input file: Input file: Downsampling association files<br>
</strong>Evaluated metrics: Complexity**<br>
<img src="external_figures/L3a2_281122/Downsampling_Barcodes_per_cCRE.svg" width="49%" /><img src="external_figures/Max_MPRA_run2/Downsampling_Barcodes_per_cCRE.svg" width="49%" /></p>
<p><strong>Legend:</strong> <br>
<strong>Interpretation:</strong><br></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="a-guide-for-running-the-analyses-notebook.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="activity-qc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/11-only_figs_associations.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
